src/proto/message_translation/power_frame_msg.hpp

```diff
@@ -0,0 +1,218 @@
+ */
+TbotsProto_PowerControl createNanoPbPowerControl(
+    ChickerCommandMode chicker_command, float kick_speed_m_per_s,
+    float chip_distance_meters, AutoChipOrKickMode auto_chip_or_kick,
+    float autochip_distance_meters, float autokick_speed_m_per_s, float angle_deg,
+    float rotation_speed_rpm, TbotsProto_PowerControl_ChargeMode charge_mode)
+{
+    TbotsProto_PowerControl control = TbotsProto_PowerControl_init_default;
+    TbotsProto_PowerControl_ChickerControl chicker =
+        TbotsProto_PowerControl_ChickerControl_init_default;
+    switch (chicker_command)
+    {
+        case ChickerCommandMode::CHIP:
+            control.chicker.which_chicker_command =
+                TbotsProto_PowerControl_ChickerControl_chip_distance_meters_tag;
+            chicker.chicker_command.chip_distance_meters = chip_distance_meters;
+            break;
+        case ChickerCommandMode::KICK:
+            control.chicker.which_chicker_command =
+                TbotsProto_PowerControl_ChickerControl_kick_speed_m_per_s_tag;
+            chicker.chicker_command.kick_speed_m_per_s = kick_speed_m_per_s;
+            break;
+        case ChickerCommandMode::AUTOCHIPORKICK:
+            control.chicker.which_chicker_command =
+                TbotsProto_PowerControl_ChickerControl_auto_chip_or_kick_tag;
+            switch (auto_chip_or_kick)
+            {
+                case AutoChipOrKickMode::AUTOCHIP:
+                    chicker.chicker_command.auto_chip_or_kick.which_auto_chip_or_kick =
+                        TbotsProto_AutoChipOrKick_autochip_distance_meters_tag;
+                    chicker.chicker_command.auto_chip_or_kick.auto_chip_or_kick
+                        .autochip_distance_meters = autochip_distance_meters;
+                    break;
+                case AutoChipOrKickMode::AUTOKICK:
+                    chicker.chicker_command.auto_chip_or_kick.which_auto_chip_or_kick =
+                        TbotsProto_AutoChipOrKick_autokick_speed_m_per_s_tag;
+                    chicker.chicker_command.auto_chip_or_kick.auto_chip_or_kick
+                        .autokick_speed_m_per_s = autokick_speed_m_per_s;
+                    break;
+                default:
+                    break;
+            }
+            break;
+        default:
+            break;
+    }
+
+    TbotsProto_PowerControl_GenevaControl geneva =
+        TbotsProto_PowerControl_GenevaControl_init_default;
+    geneva.angle_deg          = angle_deg;
+    geneva.rotation_speed_rpm = rotation_speed_rpm;
+
+    control.chicker     = chicker;
+    control.geneva      = geneva;
+    control.charge_mode = charge_mode;
+
+    return control;
+}
+/**
+ * Converts a nanopb power status msg to its google protobuf representation
+ *
+ * @param status nanopb message to convert
+ * @return a google protobuf power status matching provided nanopb
+ */
+std::unique_ptr<TbotsProto::PowerStatus> createTbotsPowerStatus(
```

Comment:
why can't the functions in primitive_google_to_nanopb_converter.cpp be used?

Reply:
I don't believe there are any functions for nanopb -> google proto in that file. They also don't have a shared parent type.